onnx/defs/op_annotation.h

```diff
@@ -0,0 +1,196 @@
+// Copyright (c) Facebook Inc. and Microsoft Corporation.
+// Licensed under the MIT license.
+
+#pragma once
+
+#include <memory>
+#include <unordered_map>
+#include <unordered_set>
+
+#include "onnx/common/assertions.h"
+
+namespace ONNX_NAMESPACE {
+
+struct EnumHash {
+  template <typename T>
+  std::size_t operator()(T t) const {
+    return static_cast<std::size_t>(t);
+  }
+};
+
+enum OpAnnotationFlag : uint16_t {
+  ElementwiseAny,
+  ElementwiseIndependent,
+  ElementwiseDependent,
+  ElementwiseStrictMonotonicIncreasing,
+  ElementwiseStrictMonotonicDecreasing,
+  ElementwiseWeakMonotonicIncreasing,
+  ElementwiseWeakMonotonicDecreasing,
+  Reduction,
+  ShapeTransform
+};
+
+using OpAnnotationFlagSet = std::unordered_set<OpAnnotationFlag, EnumHash>;
+
+class OpAnnotationHash;
+
+class OpAnnotation {
```

Comment:
Add some comments about the purpose of OpAnnotation

Reply:
Done.